app/models/concerns/orchestration/compute.rb

```diff
@@ -102,19 +103,51 @@ def delUserData
   def setComputeDetails
     if vm
       attrs = compute_resource.provided_attributes
-      normalize_addresses if attrs.keys.include?(:mac) or attrs.keys.include?(:ip)
+#      normalize_addresses if attrs.keys.include?(:mac) or attrs.keys.include?(:ip)
+
+      # mac and ip are properties of the NIC, and there may be more than one,
+      # so we need to loop. First store the nics returned from Fog in a local
+      # array so we can delete from it
+      fog_nics = vm.interfaces.dup
 
       attrs.each do |foreman_attr, fog_attr |
-        # we can't ensure uniqueness of #foreman_attr using normal rails validations as that gets in a later step in the process
-        # therefore we must validate its not used already in our db.
-        value = vm.send(fog_attr)
-        value ||= find_address if foreman_attr == :ip
-        self.send("#{foreman_attr}=", value)
-
-        if value.blank? or (other_host = Host.send("find_by_#{foreman_attr}", value))
-          delCompute
-          return failure("#{foreman_attr} #{value} is already used by #{other_host}") if other_host
-          return failure("#{foreman_attr} value is blank!")
+        if foreman_attr == :mac
```

Comment:
Can we extract this to a method? Reading the loop with so many comments and levels of indentation takes a long time for the first time at least.